services/historic_prices.go

```diff
@@ -41,12 +88,18 @@ func updateHistoricPrices() error {
 		currentDayTrunc := currentDay.Truncate(time.Hour * 24)
 		if !datesMap[currentDayTrunc.Format("01-02-2006")] {
 			historicPrice, err := fetchHistoricPrice(currentDayTrunc)
-
 			if err != nil {
 				logger.Errorf("error retrieving historic eth prices for day %v: %v", currentDayTrunc, err)
 				currentDay = currentDay.Add(time.Hour * 24)
 				continue
 			}
+
+			if historicPrice.MarketData.CurrentPrice.Usd == 0.0 {
+				logger.Warnf("no historic eth prices for day %v", currentDayTrunc)
+				currentDay = currentDay.Add(time.Hour * 24)
+				continue
+			}
```

Comment:
While I like the inclusion of this check, I am not sure whether we should only check USD. Why not check all currencies?
By the way, on Prater it looks like many rows include an `AUD` value of 0. This would not be found by this check.

What are your thoughts on that?